tools/roll-ami

```diff
@@ -108,6 +111,25 @@ def run(options, step_size=1, launch_config=None, force=False, ignore_instances=
             info('%s - running', instance['InstanceId'])
             tmp_instances.remove(instance['InstanceId'])
 
+    container_instance_id_map = build_container_instance_map(options.cluster)
+
+    # Drain old instances
+    ecs.update_container_instances_state(
+        cluster=options.cluster,
+        containerInstances=[container_instance_id_map[instance] for instance in old_instances],
+        status='DRAINING')
+
+    draining_instances = copy.copy(old_instances)
+    while draining_instances:
+        time.sleep(1)
```

Comment:
Should we wait a little longer (like 5s) to avoid hammering the ECS API, AWS has ridiculously low rate limits on a lot of their endpoints.